pilot/pkg/model/push_context.go

```diff
@@ -601,11 +609,18 @@ func (ps *PushContext) initServiceRegistry(env *Environment) error {
 	allServices := sortServicesByCreationTime(services)
 	for _, s := range allServices {
 		ns := s.Attributes.Namespace
-		switch s.Attributes.ConfigScope {
-		case networking.ConfigScope_PRIVATE:
-			ps.privateServicesByNamespace[ns] = append(ps.privateServicesByNamespace[ns], s)
-		default:
-			ps.publicServices = append(ps.publicServices, s)
+		if len(s.Attributes.ExportTo) == 0 {
+			if ps.defaultServiceExportTo[VisibilityPrivate] {
+				ps.privateServicesByNamespace[ns] = append(ps.privateServicesByNamespace[ns], s)
+			} else if ps.defaultServiceExportTo[VisibilityPublic] {
+				ps.publicServices = append(ps.publicServices, s)
+			}
```

Comment:
It does not handle explicitly specified export namespaces , like [ns1, ns2, ...]